rfc/0000-batch-job-groups.rst

```diff
@@ -0,0 +1,1179 @@
+========================
+Job Groups in Hail Batch
+========================
+
+.. author:: Jackie Goldstein
+.. date-accepted:: Leave blank. This will be filled in when the proposal is accepted.
+.. ticket-url:: Leave blank. This will eventually be filled with the ticket URL which will track the progress of the implementation of the feature.
+.. implemented:: Leave blank. This will be filled in with the first Hail version which implements the described feature.
+.. header:: This proposal is `discussed at this pull request <https://github.com/hail-is/hail-rfc/pull/0>`_. **After creating the pull request, edit this file again, update the number in the link, and delete this bold sentence.**
+.. sectnum::
+.. contents::
+.. role:: python(code)
+
+Hail Batch allows users to execute containerized workflows on a
+multi-tenant compute cluster. Users submit a single batch which
+consists of the set of job specifications to execute and the
+dependencies between jobs. A batch can be dynamically updated with
+additional jobs. However, there is no notion of structure within a
+batch. Therefore, we propose adding a new feature in Batch which
+allows users to organize jobs into groups that can be referenced
+together for key operations such as computing the status, billing
+information, and cancellation. The main motivating use case for this
+feature is Query on Batch (QoB). QoB needs more fine-grained
+cancellation abilities in order to improve the developer experience as
+well as avoid doing unnecessary work after a failure occurs in order
+to be comparable in cost to Query on Spark (QoS). The main challenges
```

Comment:
CORRECTION: In the current system, we can determine the status of a wave in O(1) requests because a Batch always has at most one Query driver. So this RFC is really about cancel after n failures.

We should discuss if we should include the ability to have more than one Query driver as a necessary motivation.